ios/Plugin/Plugin.swift

```diff
@@ -595,4 +689,76 @@ public class RadarPlugin: CAPPlugin, RadarDelegate {
         }
     }
 
+    @objc func setLogLevel(_ call: CAPPluginCall) {
+        DispatchQueue.main.async {
+            guard let levelStr = call.getString("level")?.lowercased() else {
+                call.reject("level is required")
+                return
+            }
+
+            let level: RadarLogLevel
+
+            switch levelStr {
+            case "none":
+                level = .none
+            case "error":
+                level = .error
+            case "warning":
+                level = .warning
+            case "debug":
+                level = .debug
+            default:
+                level = .info
+            }
+
+            Radar.setLogLevel(level)
+            call.resolve()
+        }
+    }
+
+}
+
+// MARK: - Internal extensions
+
+extension RadarStatus {
+
+    /// Shorthand for `Radar.stringForStatus(:)`. This is the equivalent of
+    /// `rawValue`, if `RadarStatus` were a Swift `enum`.
+    var stringValue: String {
+        return Radar.stringForStatus(self)
+    }
+
+}
+
+extension CLLocation {
+
+    /// Shorthand for `Radar.dictionaryForLocation(:)`.
+    var dictionaryValue: [AnyHashable: Any] {
+        return Radar.dictionaryForLocation(self)
+    }
+
+}
+
+extension RadarLocationSource {
+
+    // Shorthand for `Radar.stringForLocationSource(:)`. This is the equivalent
+    // of `rawValue`, if `RadarLocationSource` were a Swift `enum`.
+    var stringValue: String {
+        return Radar.stringForLocationSource(self)
+    }
+
+}
+
+extension RadarTripOptions {
+
+    /// The native SDK doesn't yet set `scheduledArrivalAt` or
+    /// `approachingThreshold` from a dictionary, so do those here, for now.
+    static func tripOptionsFromJSDictionary(_ jsDictionary: [AnyHashable: Any]) -> RadarTripOptions {
+        let options = RadarTripOptions(from: jsDictionary)
+        options.scheduledArrivalAt = jsDictionary["scheduledArrivalAt"] as? Date
+        options.approachingThreshold = UInt8(jsDictionary["approachingThreshold"] as? Int ?? 0)
```

Comment:
I am not familiar with this, but will this override the approaching threshold to 0? if so def don't want that/want to keep this null/unset

Reply:
Yes, you're right.

Reply:
Fixed in commit #15d960.